src/service.c

```diff
@@ -437,6 +437,10 @@ static void r_on_bus_acquired(GDBusConnection *connection,
 		g_error_free(ierror);
 	}
 
+	r_installer_set_compatible(r_installer, r_context()->config->system_compatible ? r_context()->config->system_compatible : NULL);
+	r_installer_set_variant(r_installer, r_context()->config->system_variant ? r_context()->config->system_variant : NULL);
+	r_installer_set_bootslot(r_installer, r_context()->bootslot ? r_context()->bootslot : NULL);
```

Comment:
The expression consisting of the ternary operator is equivalent to its first argument, hence why do you use it here?